src/network/networkd-address.c

```diff
@@ -723,10 +732,10 @@ int config_parse_address(const char *unit,
         if (streq(section, "Network")) {
                 /* we are not in an Address section, so treat
                  * this as the special '0' section */
-                section_line = 0;
-        }
+                r = address_new_static(network, "Network", 0, &n);
```

Comment:
No clue . `[Network]` is special. It holds all other sections. We want to get `struct Network` every time irrespective of dropin files.